pkg/storage/chunk/client/azure/blob_storage_client.go

```diff
@@ -105,18 +97,20 @@ func (c *BlobStorageConfig) RegisterFlags(f *flag.FlagSet) {
 // RegisterFlagsWithPrefix adds the flags required to config this to the given FlagSet
 func (c *BlobStorageConfig) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet) {
 	f.StringVar(&c.Environment, prefix+"azure.environment", azureGlobal, fmt.Sprintf("Azure Cloud environment. Supported values are: %s.", strings.Join(supportedEnvironments, ", ")))
-	f.StringVar(&c.ContainerName, prefix+"azure.container-name", "cortex", "Name of the blob container used to store chunks. This container must be created before running cortex.")
-	f.StringVar(&c.AccountName, prefix+"azure.account-name", "", "The Microsoft Azure account name to be used")
+	f.StringVar(&c.StorageAccountName, prefix+"azure.account-name", "", "Azure storage account name.")
+	f.Var(&c.StorageAccountKey, prefix+"azure.account-key", "Azure storage account key.")
+	f.StringVar(&c.ContainerName, prefix+"azure.container-name", "cortex", "Name of the storage account blob container used to store chunks. This container must be created before running cortex.")
+	f.StringVar(&c.Endpoint, prefix+"azure.endpoint-suffix", "", "Azure storage endpoint suffix without schema. The storage account name will be prefixed to this value to create the FQDN.")
+	f.BoolVar(&c.UseManagedIdentity, prefix+"azure.use-managed-identity", false, "Use Managed Identity to authenticate to the Azure storage account.")
+	f.StringVar(&c.UserAssignedID, prefix+"azure.user-assigned-id", "-", "User assigned identity ID to authenticate to the Azure storage account.")
```

Comment:
Shouldn't the default be an empty string?
```suggestion
	f.StringVar(&c.UserAssignedID, prefix+"azure.user-assigned-id", "", "User assigned identity ID to authenticate to the Azure storage account.")
```

Reply:
This looks like the default of the one I copied, I'll fix it.